controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -443,6 +448,158 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(Socket *sd,
+                        LOG(INFO) << "selected_channel=" << channel_rx;
+
+                        // send once channel preference report for each ruid
+                        auto channel_preference_tlv_tx =
+                            cmdu_tx.addClass<wfa_map::tlvChannelPreference>();
+                        if (!channel_preference_tlv_tx) {
+                            LOG(ERROR) << "addClass ieee1905_1::tlvChannelPreference has failed";
+                            return false;
+                        }
+
+                        channel_preference_tlv_tx->radio_uid() = ruid;
+
+                        // Create operating class object
+                        auto op_class_channels_tx =
+                            channel_preference_tlv_tx->create_operating_classes_list();
+                        if (!op_class_channels_tx) {
+                            LOG(ERROR) << "create_operating_classes_list() has failed!";
+                            return false;
+                        }
+
+                        // Fill operating class object
+                        op_class_channels_tx->operating_class() = operating_class;
+
+                        // allocate 1 channel
+                        if (!op_class_channels_tx->alloc_channel_list()) {
+                            LOG(ERROR) << "alloc_channel_list() has failed!";
+                            return false;
+                        }
+                        auto channel_idx   = op_class_channels_tx->channel_list_length();
+                        auto channel_tuple = op_class_channels_tx->channel_list(channel_idx - 1);
+                        if (!std::get<0>(channel_tuple)) {
+                            LOG(ERROR) << "getting channel entry has failed!";
+                            return false;
+                        }
+                        auto &channel_tx = std::get<1>(channel_tuple);
+                        channel_tx       = channel_rx;
+
+                        op_class_channels_tx->flags() = op_class_channels_rx.flags();
+                    }
+
+                    // add comma if not last channel in the list, else close list by add curl brackets
+                    if ((ch_idx + 1) != channel_list_length) {
+                        ss << ',';
+                    } else {
+                        ss << "}, ";
+                    }
+                }
+            }
+        } else if (tlvType == int(wfa_map::eTlvTypeMap::TLV_RADIO_OPERATION_RESTRICTION)) {
+            // parse radio operation restriction tlv
+            auto radio_operation_restriction_tlv_rx =
+                cmdu_rx.addClass<wfa_map::tlvRadioOperationRestriction>();
+            if (!radio_operation_restriction_tlv_rx) {
+                LOG(ERROR) << "addClass wfa_map::tlvRadioOperationRestriction has failed";
+                return false;
+            }
+
+            // TODO: continute to parse this tlv
```

Comment:
please elaborate in the TODO comment exactly what parsing is missing

Reply:
What do you mean by "what exactly"? the tlv itself of course.

Reply:
I mean that you should write here a summary of which fields in the TLV needs to be filled, and explain why they are not set in this commit.

Reply:
I mean why aren't you parsing it as part of this commit? What parsing is needed and what should be done with the parsed values? If you're not doing it now, you need to add a TODO which explains it and possible add a link to the issue which describes the whole thing.